ext/afform/core/Civi/Api4/Action/CustomGroup/GetAfforms.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+namespace Civi\Api4\Action\CustomGroup;
+
+use CRM_Afform_ExtensionUtil as E;
+
+/**
+ * Get dynamic afforms for a custom group.
+ *
+ * For single value custom groups we generate:
+ * - a field block with all of the fields from the group
+ * - a submission form for editing these fields on a parent entity record
+ *
+ * For multi-value custom groups we generate:
+ * - a field block with all of the fields from the group
+ * - a submission form for adding a new Custom record to a parent entity record
+ * - a submission form for editing a particular Custom record
```

Comment:
I think these 2 could be the same form actually.
`entity_id` could be a hidden field passed through the url.

Reply:
ah yes I see what you mean, good point

Reply:
I've kept separate forms - I think it makes sense that there's a clear route for the Create action vs Update - because `entity_id` is required for the former, but can be derived from the `id` (ie the custom record id) in the latter case.

Whilst at this point the entity_id is hidden on both forms, I would like to keep open the door for the `entity_id` being shown in future - probably readonly on the Update form; and potentially as an EntityRef select on the Create form.

I have combined the smarty templates for now though.